File: Program32.c
///////////////////////////////////////////////////////
//
//  File name :     program26.c
//  Descreption :   Factoria;
//  Author :        Vaibhav
//  Date :          11/05/2025
//
///////////////////////////////////////////////////////

#include<stdio.h>
#define ERR_INVALID -1


unsigned long Factorial(unsigned int iValue)
{
    int iCnt = 0;
    unsigned long iFact = 1;

    if(iValue < 0)
    {
        return ERR_INVALID;
    }
    for(iCnt = iValue;iCnt >= 1;iCnt--)
    {
        iFact = iFact * iCnt;
    }
    return iFact;
}

int main()
{
    unsigned int iValue = 0;
    unsigned long iRet = 0;

    printf("_____________________________________________________________________\n");
    printf("--------- This program is used for Factorial ---------\n");
    printf("_____________________________________________________________________\n");

    printf("Enter a number : ");
    scanf("%u",&iValue);

    iRet = Factorial(iValue);

    if(iRet == ERR_INVALID)
    {
        printf("Invalid Input..!");
    }
    else
    {
        printf("Factorial is : %lu\n",iRet);
    }
    printf("\n_____________________________________________________________________");
    return 0;
}

File: Java/program32.java
///////////////////////////////////////////////////////
//
//  File name :     program32.java
//  Descreption :   Factorial;
//  Author :        Vaibhav
//  Date :          20/05/2025
//
///////////////////////////////////////////////////////

import java.util.Scanner;

class Factorial
{   
    public static final int ERR_INVALID = -1;
    public int Fact(int iNo)
    {   
        int iCnt = 0;
        int iFact = 1;

        if(iNo < 0)
        {
            return ERR_INVALID;
        }
        for(iCnt = 1;iCnt <= iNo;iCnt++)
        {
            iFact = iFact * iCnt;
        }
        return iFact;
    }
}

class program32
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);
        int iValue = 0;
        int iRet = 0;

        System.out.println("_____________________________________________________________________");
        System.out.println("--------- This program is used for Factorial ---------");
        System.out.println("_____________________________________________________________________");

        System.out.println("Enter the number: ");
        iValue = sobj.nextInt();         
        
        Factorial dobj = new Factorial();
        iRet = dobj.Fact(iValue);

        if (iRet == Factorial.ERR_INVALID)
        {
            System.out.println("Invalid input");
        }
        else
        {
            System.out.println("Factorial is  "+iRet);
        }
    }
}